File: cpc21c1p1.cpp
#pragma GCC optimize("Ofast", "unroll-loops")

#include <iostream>
#include <iomanip>
#include <vector>
#include <deque>
#include <algorithm>
#include <numeric>
#include <cmath>
#include <queue>
#include <functional>
#include <stack>
#include <map>
#include <cstring>
#include <set>
#include <unordered_map>
#include <unordered_set>

#define fastio() cin.tie(0);ios::sync_with_stdio(0);
#define pii pair<int, int>
#define fi first
#define se second
#define pll pair<long long, long long>
#define mp make_pair
#define el "\n"
#define pq priority_queue
#define ll long long
#define ld long double
#define mt make_tuple
#define vi vector<int>
#define pb push_back
#define eb emplace_back
#define dq deque
#define uset unordered_set
#define umap unordered_map

using namespace std;

const int NINF = -0x3f3f3f3f;
const int INF = 0x3f3f3f3f;
const int MOD = 1e9 + 7;

//---------------End of template-------------------------------

int t;


pii prime(ll num) {
    int two = 0;
    int five = 0;

    while (num % 2 == 0) {
        num /= 2;
        two++;
    }

    while (num % 5 == 0) {
        num /= 5;
        five++;
    }

    if (num != 1) return {-1, -1};

    return {two, five};
}


int main() {
    fastio();

    cin >> t;

    while (t--) {
        ll a, b; cin >> a >> b;
        ll gcf = gcd(a, b);
        b /= gcf;
        pii f = prime(b);
        if (f.fi != -1) cout << max(f.fi, f.se) << el;
        else cout << -1 << el;
    }

}

File: cpc21c1p1.py
from math import gcd
from sys import stdin

input = stdin.readline

n = int(input())

def prime_fact(num):
    oth = 0
    two = 0
    five = 0
    i = 2
    while i * i <= num:
        if num % i:
            i += 1
        else:
            num //= i
            if i == 2: two += 1
            elif i == 5: five += 1
            else:
                return 1, 0, 0
    if num == 2: two += 1
    elif num == 5: five += 1
    else: oth += 1
    return oth, two, five


for _ in range(n):
    a, b = map(int, input().split())
    gcf = gcd(a, b)
    a //= gcf
    b //= gcf
    oth, t, f = prime_fact(b)
    if oth == 0:
        print(max(t, f))
    else:
        print(-1)
